keps/sig-scheduling/5732-topology-aware-workload-scheduling/README.md

```diff
@@ -0,0 +1,990 @@
+    // with higher scores generally indicating more preferable Placements.
+    // Plugins can implement various scoring strategies, such as bin packing to minimize
+    // resource fragmentation.
+    ScorePlacement(ctx context.Context, state *framework.CycleState, podGroup *PodGroupInfo, podSet *PodSetInfo, placement *Placement, podsAssignment *PodSetAssignment) (float64, *framework.Status)
+}
+```
+
+### Scheduling Algorithm Phases
+
+The algorithm proceeds in three main phases for a given Workload/PodGroup.
+
+#### Phase 1: Candidate Placement Generation
+
+- **Input:** PodGroupInfo and PodSetInfo.
+
+- **Action:** Iterate over distinct values of the topology label (TAS) or
+  available ResourceSlices (DRA).
+
+- **Output:** A list of Placement objects.
+
+- Example: If the label is rack, placements are generated for rack-1, rack-2,
+  etc.
+
+#### Phase 2: Pod-Level Filtering and Feasibility Check
+
+- **Action:** For each generated Placement:
+
+  1. Call `AssumePlacement` (binds context to the specific node selector/DRA
+     resources).
+
+  2. Iterate through every pod in the PodSet.
+
+  3. Run standard Pod-level Filter and Score.
+
+  4. Use internal logic to simulate placing the pod on a node.
+
+  5. If all pods fit, the Placement is marked Feasible.
+
+  6. Call `RevertPlacement`.
+
+- **Potential Optimization:** Pre-filtering can check aggregate resources
+  before running the full simulation.
+
+- **Heterogeneous PodGroup Handling**: Sequential Processing will be used
+  initially. Pods are processed sequentially; if any fail, the placement is
+  rejected.
+
+#### Phase 3: Placement Scoring and Selection
+
+- **Action:** Call `ScorePlacement` for all feasible placements.
+
+- **Selection:** Select the Placement with the highest score.
+
+- **Binding:** Proceed to bind pods to the assigned nodes and resources.
+
+### Scheduler Plugins
+
+**TopologyPlacementPlugin (New)** Implements `PlacementGenerator`. Generates
+Placements based on distinct values of the designated node label (TAS) .
+
+**DRAPlugin (Extension)** Extended to implement `PlacementGenerator` and
+`PlacementState`.
+
+- **Generator:** Returns Placements derived from available ResourceSlices
+  satisfying shared claims.
+
+- **State:** Temporarily assigns AllocationResults to ResourceClaims during
+  the Assume phase.
+
+**PlacementBinPackingPlugin (New)** Implements `PlacementScorer`. Scores
+Placements to maximize utilization (tightest fit) and minimize fragmentation.
+
+### Potential Future Extensions (Beta Candidates)
+
+The following features are out of scope for the initial Alpha implementation but
+are considered for future releases (post-1.36):
+
+1. **Prioritized Placement Scheduling:** Allowing a set of preferred placements
+   with fallbacks (e.g., prefer Rack, fallback to Block). This would introduce
+   a Rank field to the Placement struct.
+
+2. **Optional/Preferred Scheduling Constraints:** Constraints that serve purely
```

Comment:
I would make it out-of-scope for this KEP even for further releases - we can do that in a follow-up KEP if needed.